dotnet/src/Connectors/Connectors.AzureAIInference/Settings/AzureAIInferenceChatExecutionSettings.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Text.Json;
+using Azure.AI.Inference;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureAIInference;
+
+/// <summary>
+/// Chat completion execution settings.
+/// </summary>
+public class AzureAIInferenceChatExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureAIInferenceChatExecutionSettings"/> class.
+    /// </summary>
+    public AzureAIInferenceChatExecutionSettings()
+    {
+        this.ExtensionData = new Dictionary<string, object>();
+    }
+
+    /// <summary>
+    /// Allowed values: "error" | "drop" | "pass-through"
+    /// </summary>
+    public string? ExtraParameters
+    {
+        get => this.GetValueExtensionData("extra_parameters") as string;
+        set
+        {
+            this.ThrowIfFrozen();
+            this.SetValueInExtensionData(value!, "extra_parameters");
```

Comment:
We don't duplicate the value in extension data elsewhere, why are we doing it here?

The purpose of extension data was to allow for deserialization from prompt execution settings in our YAML file format. But the AI Service will always convert to the type of prompt execution settings it understands before using the values.

Reply:
OK, got it.

But this makes the implementation inconsistent and possibly introduces the possibility of subtle errors if some sets the wrong data type in the extension data.

We should discuss changing the pattern and then do it across all connectors rather than introducing different behaviours for different connectors.

Reply:
According to our offline discussion, this changes will be discussed later, as we have the new abstractions in place.